File: server/CSM Gate Foundation Server/CSM Gate Foundation Core/Services/Models/Inputs/AuthInput.cs
namespace CSM_Gate_Foundation_Core.Services.Models.Inputs;

/// <summary>
///     Represents an authentication request data input.
/// </summary>
public record AuthInput {

    /// <summary>
    ///     Solution signature to authenticate.
    /// </summary>
    public required string Sign { get; set; }

    /// <summary>
    ///     User identity.
    /// </summary>
    public required string Username { get; init; }

    /// <summary>
    ///     User secret password.
    /// </summary>
    public required byte[] Password { get; init; }
}

File: client/csm_gate_foundation_client/lib/src/models/inputs/auth_input.dart
import 'dart:convert';
import 'dart:typed_data';

import 'package:csm_client_core/csm_client_core.dart';

/// Represents an { Auth } operation input data.
final class AuthInput implements IEncodable {
  /// Solution signature.
  String sign = '';

  /// User account identity.
  String username = '';

  /// User account password sign.
  Uint8List password = Uint8List(0);

  /// Generates a new [AuthInput] object.
  AuthInput();

  ///
  AuthInput.a(
    this.sign,
    this.username,
    this.password,
  );

  @override
  DataMap encode() {
    return <String, dynamic>{
      'sign': sign,
      'username': username,
      'password': base64Encode(password),
    };
  }
}
